File: comap_server/entities/ImageStyle.py
# entities/image_style.py
import uuid
from sqlalchemy import Column, String, Boolean, Integer
from database import Base


class ImageStyleEntity(Base):
    __tablename__ = 'image_styles'

    # Its own GUID
    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))

    # Link to the Project (We will make this a Foreign Key later when Project is ready)
    projectId = Column(String, nullable=False, index=True)

    # Band Configurations
    singleBand = Column(Boolean, default=False)
    band1 = Column(String, nullable=True)
    band2 = Column(String, nullable=True)
    band3 = Column(String, nullable=True)

    # Adjustments
    brightness = Column(Integer, nullable=True)
    contrast = Column(Integer, nullable=True)
    hue = Column(Integer, nullable=True)
    saturation = Column(Integer, nullable=True)
    lightness = Column(Integer, nullable=True)

    # Levels
    autoLevel = Column(Boolean, default=False)
    saturateLevel = Column(Boolean, default=False)
    saturationValue = Column(Integer, nullable=True)

File: comap_app/src/models/image-style.js
export class ImageStyle {
    constructor(data = {}) {
        this.projectId = data.projectId || null;

        // Bands
        this.singleBand = data.singleBand || null; // If set, grayscale mode
        this.band1 = data.band1 || "B04"; // Red
        this.band2 = data.band2 || "B03"; // Green
        this.band3 = data.band3 || "B02"; // Blue

        // Correction
        this.brightness = data.brightness || 0;
        this.contrast = data.contrast || 0;
        this.hue = data.hue || 0;
        this.saturation = data.saturation || 0;
        this.lightness = data.lightness || 0;

        // Histogram
        this.autoLevel = data.autoLevel || false;
        this.saturateLevel = data.saturateLevel || false;
        this.saturationValue = data.saturationValue || 2.0; // e.g. 2% cut
    }
}
